d2renderers/d2latex/latex.go

```diff
@@ -0,0 +1,79 @@
+//go:build cgo
+
+package d2latex
+
+import (
+	_ "embed"
+	"fmt"
+	"math"
+	"regexp"
+	"strconv"
+
+	"oss.terrastruct.com/xdefer"
+	v8 "rogchap.com/v8go"
+)
+
+//go:embed polyfills.js
+var polyfillsJS string
+
+//go:embed setup.js
+var setupJS string
+
+//go:embed mathjax.js
+var mathjaxJS string
+
+var svgRe = regexp.MustCompile(`<svg[^>]+width="([0-9\.]+)ex" height="([0-9\.]+)ex"[^>]+>`)
+
+func Render(s string) (_ string, err error) {
+	defer xdefer.Errorf(&err, "latex failed to parse")
+	v8ctx := v8.NewContext()
+
+	if _, err := v8ctx.RunScript(polyfillsJS, "polyfills.js"); err != nil {
+		return "", err
+	}
+
+	if _, err := v8ctx.RunScript(mathjaxJS, "mathjax.js"); err != nil {
+		return "", err
+	}
+
+	if _, err := v8ctx.RunScript(setupJS, "setup.js"); err != nil {
+		return "", err
+	}
+
+	val, err := v8ctx.RunScript(fmt.Sprintf(`adaptor.innerHTML(html.convert("%s", {
+  em: 16,
+  ex: 8,
+}))`, s), "value.js")
+	if err != nil {
+		return "", err
+	}
+
+	return val.String(), nil
+}
+
+func Measure(s string) (width, height int, err error) {
+	defer xdefer.Errorf(&err, "latex failed to parse")
+	svg, err := Render(s)
+	if err != nil {
+		return 0, 0, err
+	}
+
+	dims := svgRe.FindAllStringSubmatch(svg, -1)
+	if len(dims) != 1 || len(dims[0]) != 3 {
+		return 0, 0, fmt.Errorf("svg parsing failed for latex: %v", svg)
+	}
+
+	wEx := dims[0][1]
+	hEx := dims[0][2]
+
+	wf, err := strconv.ParseFloat(wEx, 64)
+	if err != nil {
+		return 0, 0, fmt.Errorf("svg parsing failed for latex: %v", svg)
```

Comment:
not `sag parsing failed`, `float parsing failed` and then you can use `wEx` and `hEx` or you can get rid of them

Reply:
float parsing must've failed because svg parsing failed though, it's a more informative error statement